registry/storage/driver/gcs/gcs.go

```diff
@@ -59,6 +60,44 @@ type driverParameters struct {
 	projectID     string
 }
 
+// NewDriverParameters creates and fills in a new DriverParamters struct with the values given.
+// It expects that jsonKey is a byte slice representing the JWT key for the GCS account.
+// If it is specified as nil, this function tries to use the default token source as the token
+func NewDriverParameters(bucket string, rootDir string, jsonKey []byte) (*DriverParameters, error) {
+	params := new(DriverParameters)
+	params.bucket = bucket
+	params.rootDirectory = rootDir
+
+	jwtConf := new(jwt.Config)
+	var ts oauth2.TokenSource
+	var key struct {
+		ProjectID string `json:"project_id"`
+	}
+
+	if jsonKey != nil {
+		jwtConf, err = google.JWTConfigFromJSON(jsonKey, storage.ScopeFullControl)
```

Comment:
This needs to be assigned with `:=` since `err` hasn't appeared before in this function.